featurebyte/api/feature_group.py

```diff
@@ -495,9 +483,31 @@ def save(self, conflict_resolution: ConflictResolution = "raise") -> None:
         ... ])
         >>> features.save()  # doctest: +SKIP
         """
-        # Raise error if the feature has been saved (as feature group saving equals to saving feature individually).
-        # Save an already saved feature on Feature level will raise the same ObjectHasBeenSavedError.
-        self._batch_feature_save(
+        # prepare batch feature create payload
+        feature_list_batch_feature_create = self._get_feature_list_batch_feature_create_payload(
+            feature_list_id=ObjectId(),
+            feature_list_name=f"_temporary_feature_list_{datetime.now()}",
             conflict_resolution=conflict_resolution,
-            to_raise_object_has_been_saved_error=True,
+        )
+
+        # save temporary feature list with list of features
+        self.post_async_task(
+            route="/feature_list/batch",
+            payload=feature_list_batch_feature_create.json_dict(),
+            retrieve_result=False,
+            has_output_url=False,
+        )
+        feature_list_dict = get_api_object_by_id(
```

Comment:
Would it make sense to wrap the lines below in a try block and delete the temporary feature list in the finally section to ensure clean up on unexpected errors?

Reply:
The temporary feature list is only created if all of the features have been created successfully. It is the last step of the async job. If we put it inside the try-finally block, it will becomes something like:
```
try:
    ....
finally:
    try:
         delete_api_object_by_id(route="/feature_list", id_value=feature_list_batch_feature_create.id)
    except:
         pass
```

Reply:
It may still have some value if there are some unexpected error happens in between finished async job & deletion happens, though i think the chance is small.